src/data/attribute.rs

```diff
@@ -222,6 +222,19 @@ fn process_metadata(s: &Struct, prefix: String) -> Vec<(String, String)> {
     result
 }
 
+pub fn store_request_body_size(body_size: usize) -> Result<(), PropertyError> {
+    let attr = format!("{KUADRANT_NAMESPACE}\\.internal\\.request_body_size");
+    set_attribute(attr.as_str(), (body_size as u64).to_le_bytes().as_slice())
```

Comment:
Why storing that _outside_ of the `HttpContext`/`KuadrantFilter` ? Some comment somewhere implies it needs to be made available to the CEL's `Context` as well, which also confuses me...

Reply:
When `requestBodyJSON` is invoked (at CEL eval time), there  is no access to HttpContext/KuadrantFilter. This function needs to read request body using `proxy_wasm::hostcalls::get_buffer` and it needs the body size as input arg. [here](https://github.com/Kuadrant/wasm-shim/pull/189/files#diff-5c9e9bf6e990ebde97176b69053f106a454e54ea378947a0e1faaf5e25a51580R115)

Reply:
Just flip the indirection on its head. Have the size populated from the call site

Reply:
Today you're sort of using the `metadata` storage as a "request local static" to store the body size when it's made available, then retrieve it back when you need it. But you don't need to store it on the filter's state, internal to the host, to achieve that. 

I have mentioned in the past that probably having something "representing the `Request`" is what we would need to keep that state around... and avoid going back to the host all the time, e.g. to retrieve the same header 5 times... That thing could just get the body size injected when the phase passes it to the filter (that in turn would hold the instance of that `Request`, shared with the CEL `Context` for evaluation)

Reply:
ok, that's clearer now. TBH I never liked this mechanism of using host metadata to keep data for later usage. I thought about injecting context, but the impact was way bigger than the "temporary storage" solution implemented here. Check our slack conversation on 2nd of June :wink: you might have now more context to understand what I mean at that time :smile: